src/neo/Plugins/Plugin.cs

```diff
@@ -47,176 +19,29 @@ public abstract class Plugin : IDisposable
-                    }
-                    catch (FormatException) { }
-                    break;
-                case ".dll":
-                    if (e.ChangeType != WatcherChangeTypes.Created) return;
-                    if (GetDirectoryName(e.FullPath) != PluginsDirectory) return;
-                    try
-                    {
-                        LoadPlugin(Assembly.Load(File.ReadAllBytes(e.FullPath)));
-                    }
-                    catch { }
-                    break;
-            }
-        }
-
-        private static Assembly CurrentDomain_AssemblyResolve(object sender, ResolveEventArgs args)
-        {
-            if (args.Name.Contains(".resources"))
-                return null;
-
-            AssemblyName an = new(args.Name);
-
-            Assembly assembly = AppDomain.CurrentDomain.GetAssemblies().FirstOrDefault(a => a.FullName == args.Name);
-            if (assembly is null)
-                assembly = AppDomain.CurrentDomain.GetAssemblies().FirstOrDefault(a => a.GetName().Name == an.Name);
-            if (assembly != null) return assembly;
-
-            string filename = an.Name + ".dll";
-            string path = filename;
-            if (!File.Exists(path)) path = Combine(GetDirectoryName(Assembly.GetEntryAssembly().Location), filename);
-            if (!File.Exists(path)) path = Combine(PluginsDirectory, filename);
-            if (!File.Exists(path)) path = Combine(PluginsDirectory, args.RequestingAssembly.GetName().Name, filename);
-            if (!File.Exists(path)) return null;
-
-            try
-            {
-                return Assembly.Load(File.ReadAllBytes(path));
-            }
-            catch (Exception ex)
-            {
-                Utility.Log(nameof(Plugin), LogLevel.Error, ex);
-                return null;
-            }
-        }
-
         public virtual void Dispose()
         {
         }
 
         /// <summary>
-        /// Loads the configuration file from the path of <see cref="ConfigFile"/>.
-        /// </summary>
-        /// <returns>The content of the configuration file read.</returns>
-        protected IConfigurationSection GetConfiguration()
-        {
-            return new ConfigurationBuilder().AddJsonFile(ConfigFile, optional: true).Build().GetSection("PluginConfiguration");
-        }
-
-        private static void LoadPlugin(Assembly assembly)
-        {
-            foreach (Type type in assembly.ExportedTypes)
-            {
-                if (!type.IsSubclassOf(typeof(Plugin))) continue;
-                if (type.IsAbstract) continue;
-
-                ConstructorInfo constructor = type.GetConstructor(Type.EmptyTypes);
-                try
-                {
-                    constructor?.Invoke(null);
-                }
-                catch (Exception ex)
-                {
-                    Utility.Log(nameof(Plugin), LogLevel.Error, ex);
-                }
-            }
-        }
-
-        internal static void LoadPlugins()
```

Comment:
Why these methods are removed?

Reply:
My thought is that plugin loading should be handled by the Neo dll host (i.e. neo-cli). However, I can see the value of providing helper methods like `LoadPlugins` that the neo host can choose to use if they wish. 

I opened this PR as draft to get feedback on topics like this